pecos/core/utils/clustering.hpp

```diff
@@ -31,6 +31,58 @@ enum {
     SKMEANS=5,
 }; /* partition_algo */
 
+enum {
+    CONSTANT_SAMPLE_SCHEDULE=0,
+    LINEAR_SAMPLE_SCHEDULE=1,
+    POLYNOMIAL_SAMPLE_SCHEDULE=2,
+}; /* sample strategies */
+
+extern "C" {
+    struct SampleSchedulerParam {
+        int strategy;
+        float sample_rate;
+        float warmup_sample_rate;
+        float warmup_layer_rate;
+
+        SampleSchedulerParam(
+            int strategy,
+            float sample_rate,
+            float warmup_sample_rate,
+            float warmup_layer_rate
+        ): strategy(strategy),
```

Comment:
check the input sample rate at line 159.

Reply:
Checking/validating the arguments in later stage (e.g., line 159) is not a good practice because
1. Its hard to maintain;
2. the program still work despite of the wrong input argument;
3. the program crash after running for a certain amount of time (loading data, etc).

Reply:
Is there any reason we can not do the checking at the constructor of `ClusteringSamplerParam` or  `ClusteringSampler` class?